extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/exec/ControllerImpl.java

```diff
@@ -412,9 +414,15 @@ public TaskStatus runTask(final Closer closer)
       final String selfHost = MSQTasks.getHostFromSelfNode(selfDruidNode);
       final MSQErrorReport controllerError =
           exceptionEncountered != null
-          ? MSQErrorReport.fromException(id(), selfHost, null, exceptionEncountered)
+          ? MSQErrorReport.fromException(
+              id(),
+              selfHost,
+              null,
+              exceptionEncountered,
+              task.getQuerySpec().getColumnMappings()
+          )
           : null;
-      final MSQErrorReport workerError = workerErrorRef.get();
+      MSQErrorReport workerError = mapQueryColumnNameToOutputColumnName(workerErrorRef.get());
```

Comment:
IMHO: this change should go into : 

https://github.com/apache/druid/blob/3ddcaa0180ec58ae1476f3d54610c30ace3a948e/extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/exec/ControllerImpl.java#L739-L747

Reply:
Makes much more sense, incorporated the feedback in my latest commit.